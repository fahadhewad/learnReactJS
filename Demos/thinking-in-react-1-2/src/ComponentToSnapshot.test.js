import { render, screen } from '@testing-library/react';
import ComponentToSnapshot from './ComponentToSnapshot';

xtest(`Snapshot ComponentToTest`, () => {
    expect(render(<ComponentToSnapshot />)).toMatchSnapshot();
});

describe(`ComponentToSnapshot props render tests`, () => {

    test(`title prop is rendered`, () => {
        // Arrange
        const testValue = `Test Heading`;
        render(<ComponentToSnapshot title={testValue} />);

        //Act
        const result = screen.getByText(testValue);

        // Assert
        expect(result).toBeInTheDocument();
    })
});