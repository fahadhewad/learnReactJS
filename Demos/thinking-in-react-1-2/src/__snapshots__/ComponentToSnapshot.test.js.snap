// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot ComponentToTest 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <h1>
          I'm a static, none-changing component
        </h1>
        <a
          href="https://www.bbc.co.uk"
        >
          Link to the BBC
        </a>
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <h1>
        I'm a static, none-changing component
      </h1>
      <a
        href="https://www.bbc.co.uk"
      >
        Link to the BBC
      </a>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
